src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -817,12 +892,12 @@ void ov::npuw::JustInferRequest::unpack_closure(std::size_t idx, RqPtr request)
     }
 }
 
-void ov::npuw::JustInferRequest::recreate_subrequests(std::size_t idx) {
-    auto& comp_model_desc = m_npuw_model->m_compiled_submodels[idx];
-    auto real_idx = comp_model_desc.replaced_by.value_or(idx);
+void ov::npuw::JustInferRequest::recreate_subrequests(std::size_t real_idx) {
+    auto& comp_model_desc = m_npuw_model->m_compiled_submodels[real_idx];
+    NPUW_ASSERT(comp_model_desc.replaced_by.value_or(real_idx) == real_idx);
```

Comment:
Please refer to the minimal fix here: https://github.com/openvinotoolkit/openvino/pull/27081/commits/149d1d6785d1a7931375ba9f1f235a1ba0cbd5ba

Reply:
Used it, thanks a lot!

Reply:
so does it work now if runtime failure occurs in non-1st function block? Not talking about accuracy failover here.